ledfx/color.py

```
@@ -298,6 +339,29 @@
+    """
+    Accepts a Gradient instance, an inline gradient string, or a simple color
+    string and returns a hex color sampled at `position` (0.0..1.0).
+    """
+    # If caller passed a Gradient instance, use it directly
+    if isinstance(gradient_like, Gradient):
+        return gradient_like.sample(position)
+
+    # If it is a simple RGB value returned by parse_gradient, handle
+    try:
+        parsed = parse_gradient(gradient_like)
+    except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)